test/e2e/framework/job/fixtures.go

```diff
@@ -119,7 +123,7 @@ func NewTestJobOnNode(behavior, name string, rPol v1.RestartPolicy, parallelism,
 
 // setup host path directory to pass information between pod restarts
 func setupHostPathDirectory(job *batchv1.Job) {
-	if len(job.Spec.Template.Spec.NodeName) > 0 {
+	if _, nodeNameSpecified := job.Spec.Template.Spec.NodeSelector["kubernetes.io/hostname"]; nodeNameSpecified {
```

Comment:
subtle

Reply:
Indeed, this is why I had the issue in the initial fix. However, I checked this function is only used internally from `NewTestJobOnNode` so it should be fine.